src/main/java/reactor/netty/transport/TransportConfig.java

```diff
@@ -236,12 +236,33 @@ protected TransportConfig(TransportConfig parent) {
 		this.preferNative = parent.preferNative;
 	}
 
+	/**
+	 * Return the channel type this configuration is associated with, it can be one of the following:
+	 * <ul>
+	 *   <li>{@link io.netty.channel.socket.SocketChannel}</li>
+	 *   <li>{@link io.netty.channel.socket.ServerSocketChannel}</li>
+	 *   <li>{@link io.netty.channel.unix.DomainSocketChannel}</li>
+	 *   <li>{@link io.netty.channel.unix.ServerDomainSocketChannel}</li>
+	 *   <li>{@link io.netty.channel.socket.DatagramChannel}</li>
+	 * </ul>
+	 *
+	 * @param isDomainSocket true if {@link io.netty.channel.unix.DomainSocketChannel} or
+	 * {@link io.netty.channel.unix.ServerDomainSocketChannel} is needed, false otherwise
+	 * @return the channel type this configuration is associated with
+	 */
+	protected abstract Class<? extends Channel> channelType(boolean isDomainSocket);
```

Comment:
can we expect new type of sockets to be supported at some point? (see if an enum would make sense). It seems there's a lot of dimensions now: client vs server + transport protocol + isNative + isDomainSocket. Is there a simplification possible due to mutually exclusive dimensions? (eg. can we collapse the two booleans into a single enum?)

Reply:
I'll think about that and address on a later stage.